### PR TITLE
Updated readme to streamline instruction, changed RPI setup script to not add unnecessary deps, and fixed script pre-flight checks.

### PR DESCRIPTION
**Pull request by ELLEation**
- setup_host_rpi.sh
  
  General Updates
  
      Header Updates:
          Added script metadata including script name, author, date, and description.
          Enhanced clarity regarding the purpose of the script (specific to Raspberry Pi with Spotify CarThing).
  
      Comment Improvements:
          Standardized and improved comments for better readability.
          Reformatted usage notes for functions.
  
      Enhanced Error Handling:
          Introduced checks for Raspbian Bullseye compatibility.
          Enhanced network interface detection with fallback mechanisms.
  
      Dynamic Network Interface Handling:
          Added support for dynamically identifying and selecting network adapters if multiple NICs are detected.
          Improved logic for setting the WAN_INTERFACE.
  
  Feature Additions
  
      Developer Mode:
          Added an interactive prompt to enable developer mode for remote access.
          Introduced port forwarding (e.g., SSH, VNC, debugging) if developer mode is enabled.
  
      Dependency Installation:
          Automatically installs required dependencies (iptables, iptables-persistent).
  
      System Updates:
          Added a system update process to ensure packages are up-to-date.
  
      Dev Mode Configuration:
          Conditionally forwards ports (2022, 5900, 9222) based on developer mode status.
  
  Improved IP Tables Handling
  
      Rule Management:
          Added messages to indicate when rules are being cleared, rewritten, or persisted.
          Consolidated NAT masquerading rules for enhanced clarity.
  
      Dynamic Configurations:
          Forwarding rules now dynamically apply to the selected network adapter.
  
  USB and Network Configuration
  
      Enhanced USB Detection:
          Updated logic to check for an inactive usb0 network interface using a specific IP pattern (192.168.1).
  
      Custom Configurations:
          Refined USB network settings and ensured they are saved in interfaces.d.
  
      Superbird Hostname Addition:
          Automatically appends superbird to /etc/hosts if not already present.
  
  Logging and User Interaction
  
      Progress Indicators:
          Added clear console messages for each major step to improve user feedback.
      Interactive Choices:
          Prompted user interaction for network interface selection and developer mode setup.
  
  Minor Changes
  
      Standardized function naming and descriptions.
      Minor adjustments to ensure compatibility with modern Raspberry Pi configurations.

 - README.md

  Streamlined Instructions:
  
      Consolidated detailed platform-specific setup steps into simplified instructions for flashing the CarThing.
  
  Introduction of Terbium:
  
      Added Terbium as the recommended web-based tool for flashing the CarThing, reducing reliance on the command-line superbird-tool.
  
  Revised "How to Use" Section:
  
      Simplified prerequisites for running Nocturne on CarThing.
      Expanded supported host devices to include Raspberry Pi, Windows, Mac, and Linux.
  
  Flashing Instructions Reorganized:
  
      Grouped flashing methods into collapsible sections with a preference for the Terbium method.
      Included a brief mention of using superbird-tool for advanced users.
  
  Updated Raspberry Pi Setup:
  
      Streamlined instructions for setting up Raspberry Pi with a focus on the latest tools and configuration steps.
  
  Platform-Specific Notes:
  
      Marked MacOS and Linux as "Not Recommended" with placeholders for future details.
      Provided a basic unsupported method for Windows, warning users about potential issues.
  
  Enhanced UI Setup Instructions:
  
      Directed users to follow UI setup steps in the linked Nocturne UI repository.
  
  Miscellaneous:
  
      Removed verbose, redundant instructions to improve readability.
      Updated links and clarified paths for downloading and setting up files.

### CODE REVIEW

File: README.md
Comment (BBaoVanC) [inline README.md:113]: Remember that this would need to be updated to `setup_hotspot_rpi.py` because of the rename.

File: README.md
Comment (BBaoVanC) [inline README.md:29]: nit: "Car Thing" should be two words

```suggestion
### Flashing the Spotify Car Thing
```

File: README.md
Comment (BBaoVanC) [inline README.md:36]: These manual line breaks add an unnatural spacing, I think they should be removed.

File: README.md
Comment (BBaoVanC) [inline README.md:44]:
```suggestion
```

File: README.md
Comment (BBaoVanC) [inline README.md:49]:
```suggestion
```

File: README.md
Comment (BBaoVanC) [inline README.md:53]:
```suggestion
```

File: README.md
Comment (BBaoVanC) [inline README.md:67]:
```suggestion
```

File: README.md
Comment (BBaoVanC) [inline README.md:74]:
```suggestion
```

File: README.md
Comment (BBaoVanC) [inline README.md:120]:
```suggestion
```

File: README.md
Comment (BBaoVanC) [inline README.md:105]:
```suggestion
Optionally for portable use, you can configure the Pi to create a hotspot for its internet access as follows:
```

File: README.md
Comment (BBaoVanC) [inline README.md:141]: Why add these if they aren't completed? These two sections on using Windows/macOS as a host device should be removed.

File: README.md
Comment (BBaoVanC) [inline README.md:168]:
```suggestion
```

File: README.md
Comment (BBaoVanC) [inline README.md:178]:
```suggestion
```

File: README.md
Comment (BBaoVanC) [inline README.md:168]: Have you tested this windows method? I also would need input from the team on whether we want to bother mentioning this method, as it implies some level of support (even if you say that it's unsupported).

**Review by BBaoVanC — CHANGES_REQUESTED** [review]
This PR is a good start. It brings a much needed cleanup of the README and RPi setup script. Please see my comments; many of these are nitpicks but I think it's better to mention now than leave them and fix later.

I don't have a Pi to test the new script on myself, so I will ask the team to do testing at some point before the PR is merged. And I haven't quite combed through it with a full accuracy check on everything as the old Debian image isn't what I'm most experienced with.

File: README.md
Comment (ELLEation) [reply README.md:168]: Yes I have this working, even on an AMD chip with the bad USB chipset. Occasionally it takes a few powercycles for it to be recognized. But it works a majority of the time. New AMD and Intel platforms have no issues. 

See below:

File: setup-scripts/setup_host_rpi.sh
Comment (ELLEation) [reply setup-scripts/setup_host_rpi.sh:80]: Yes this works, the problem is that before you were checking if the string "inet " was present. However because of APIPA it was getting an automatically configured IP address. Thus failing the check. This checks to see if it has been explicitly configured for the Car Thing subnet.

**Comment by ELLEation** [comment]
Check out the latest commits. It addresses a lot of the issues and nitpicks you had. I left a lot of the comments however. My general rule of thumb is more comments is better than none, especially if you're wanting to allow the end user to understand what the script does without them having to be a pro at bash/shell scripting.

**Comment by BBaoVanC** [comment]
> My general rule of thumb is more comments is better than none

My rule of thumb is that redundant comments are worse than none, especially when they can later become out of sync with the code and misleading. For example, simply saying "VARIABLES" above a section of several variables doesn't really add any new information that isn't already obvious just from looking at it. Likewise, putting "update system" immediately above `apt update`, or "Allow IP forwarding" immediately before an echo command saying "Enabling IP forwarding" doesn't really add any information that isn't obvious from the script's text.

It might not be a big deal but I think it's important to keep comments clean. If you have too many comments that feel useless, your eyes start to gloss over them. This is a problem because comments should always command attention.

I will re-review in a little bit, unless you are still actively working on something.

File: README.md
Comment (BBaoVanC) [reply README.md]: The comment also needs to be renamed, but otherwise this is fixed.

File: README.md
Comment (BBaoVanC) [reply README.md]: I'll remove this comment from my review for now and I'll ask the team to check on it later on.

**Review by BBaoVanC — COMMENTED** [review]
I will merge this and change the comments later.

Advice for the future, and to make maintainers' lives easier in general when you contribute:

- keep PRs smaller and more focused
- consult with maintainers before making huge changes or rewrites
- make a better PR summary, because the one here was pretty unreadable (I ended up just having to look at the diffs on my own)

But overall, thank you for the contribution; it does make the README a lot cleaner and easier to navigate and the setup script less of a mess.

**Pull request merged**
